ethcore/sync/src/chain/supplier.rs

```diff
@@ -232,6 +219,24 @@ impl SyncSupplier {
 		Ok(Some((BlockHeadersPacket.id(), rlp)))
 	}
 
+	/// Respond to GetPooledTransactions request
+	fn return_pooled_transactions(io: &dyn SyncIo, r: &Rlp, peer_id: PeerId) -> RlpResponseResult {
+		const LIMIT: usize = 256;
+
+		let transactions = r.iter().take(LIMIT).filter_map(|v| {
+			v.as_val::<H256>().ok().and_then(|hash| io.chain().pooled_transaction(hash))
+		}).collect::<Vec<_>>();
+
+		let added = transactions.len();
+		let mut rlp = RlpStream::new_list(added);
+		for tx in transactions {
```

Comment:
We have methods to make sure the package does not end up oversized, please take a look how we construct transaction packages during propagation I think the same logic should apply here (instead of having new arbitrary limit of 256 transactions).

Reply:
The 256 comes from the EIP and is a suggestion. Defntly should be combined with `io.payload_soft_limit()` (4Mb I think) like we do in other places.